### PR TITLE
FUSETOOLS-2060 - Always return non-null value for iconName

### PR DESCRIPTION
**Pull request by apupier**
there is another issue which is that the referenced endpoint is not resolved, but we will have the default icon instead of a NPE. Seems to be a first good step.

### CODE REVIEW

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/AbstractCamelModelElement.java [hunk 1]
```diff
@@ -1535,26 +1535,24 @@ public String getIconName() {
...
 				if (u.startsWith("ref:")) {
 					// if its a ref we lookup what is the reference scheme
-					String refId = u.substring(u.indexOf(":") + 1);
+					String refId = u.substring(u.indexOf(':') + 1);
 					AbstractCamelModelElement endpointRef = getCamelContext().getEndpointDefinitions().get(refId);
 					if (endpointRef != null) {
 						String refUri = (String) endpointRef.getParameter(URI_PARAMETER_KEY);
 						if (refUri != null) {
-							scheme = refUri.substring(0, refUri.indexOf(":"));
+							return refUri.substring(0, refUri.indexOf(':'));
```
Comment (lhein) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/AbstractCamelModelElement.java]: should we cover the case where no : is in the uri for whatsoever reason?

**Comment by lhein** [comment]
+1 on success

**Comment by apupier** [comment]
testPR

**Comment by apupier** [comment]
previous job build failed, I investigated a bit but cannot find something useful to avoid the isseu again. i think that it is not related to this PR and an "asynchrone" issue. The only stacktraces that I found suspect were already reported on Eclipse bugtracker.

**Pull request merged**